### PR TITLE
Add user guide section on producing and consuming an "outgoing artifact"

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

For more information, see https://github.com/gradle/dotorg-docs/issues/231.

This section describes one of the use cases. There may be other use cases and solution we'll want to propose. We should ping the build tool team for feedback.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc [hunk 1]
```diff
@@ -391,6 +391,29 @@ If you come from Maven land you might be perfectly happy with this. If you come
...
 
+[[sec:depending_on_output_of_another_project]]
+=== Depending on the task output produced by another project
+
+<<sec:project_jar_dependencies,Project dependencies>> model dependencies between modules. Effectively, you are saying that you depend on the main output of another project. In a Java-based project that's usually a JAR file.
+
+Sometimes you may want to depend on an output produced by another task. In turn you'll want to make sure that the task is executed beforehand to produce that very output. Declaring a task dependency from one project to another is a poor way to model this kind of relationship and introduces unnecessary coupling. The recommended way to model such a dependency is to produce the output, mark it as "outgoing" artifact or add it to the output of the `main` source set which you can depend on in the consuming project.
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc]: as _an_ outgoing artifact

File: subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc [hunk 2]
```diff
@@ -391,6 +391,29 @@ If you come from Maven land you might be perfectly happy with this. If you come
...
+++++
+
+The consuming project is supposed to be able to read the properties file at runtime. Declaring a project dependency on the producing project takes care of creating the properties beforehand and making it available to the runtime classpath.
+
+++++
+<sample id="buildInfoConsumer" dir="userguide/multiproject/dependencies/outgoingArtifact/consumer" title="Declaring a project dependency on the project producing the properties file">
+    <sourcefile file="build.gradle" snippet="producer-project-dependency"/>
+</sample>
+++++
+
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc]: I feel a closing statement like "In the example above, the consumer now declares a dependency on the outputs of the producer project" would be helpful.

File: subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc (hunk 1 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc]: done

File: subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc (hunk 2 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/userguide/multiProjectBuilds.adoc]: done

**Pull request merged**